dwio/nimble/velox/FieldReader.h

```diff
@@ -63,6 +63,13 @@ class FieldReader {
 
   virtual ~FieldReader() = default;
 
+  // Estimation of the total output size of the current reading stripe in bytes.
+  // This method will throw NotSupported error for not yet supported encoding
+  // estimations.
+  //
+  // NOTE: This is not the estimation on the remaining rows, but the total rows.
+  virtual uint64_t estimatedTotalOutputSize() const = 0;
```

Comment:
As we discussed offline, the main challenge with this is due to scattering coming from upstream nodes in the schema tree. We also established that scattering is dominated by the top level nullability of the entire row. So the approach is generally pretty sound.

It will, however, over account the variable length types with offsets and lengths, like array and maps, since the scattering doesn't apply to element soup. I'd say, if you really want to be accurate, you can return a pair/struct in this interface to distinguish variable length vs fixed length content, and apply your ratio on fixed length content only.

Reply:
Let's rename this method to totalMaterializedSize(). If you intend to distinguish the fixed length vs not, this will be quite accurate. You can leave it in the comment that it's not yet accurate right now if you don't make this improvement in this iteration.